### PR TITLE
Add new datasets for R4DS

### LINKED ISSUE #1331: Add datasets useful for R4DS
```R
who2 <- who |> 
  rename_with(~ str_remove(.x, "new_?")) |> 
  rename_with(~ str_replace(.x, "([mf])", "\\1_")) |> 
  select(!starts_with("iso"))
who2

family <- tribble(
  ~family,  ~dob_child1,  ~dob_child2, ~name_child1, ~name_child2,
        1, "1998-11-26", "2000-01-29",      "Susan",       "Jose",
        2, "1996-06-22",           NA,       "Mark",           NA,
        3, "2002-07-11", "2004-04-05",        "Sam",       "Seth",
        4, "2004-10-10", "2009-08-27",      "Craig",       "Khai",
        5, "2000-12-05", "2005-02-28",     "Parker",     "Gracie",
)
family <- family %>% 
  mutate(across(starts_with("dob"), parse_date))
family
```

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1331

### CODE REVIEW

File: data-raw/family.R [hunk 1]
```diff
@@ -0,0 +1,15 @@
+library(dplyr)
+
+family <- tribble(
+  ~family,  ~dob_child1,  ~dob_child2, ~name_child1, ~name_child2,
+        1, "1998-11-26", "2000-01-29",      "Susan",       "Jose",
+        2, "1996-06-22",           NA,       "Mark",           NA,
+        3, "2002-07-11", "2004-04-05",        "Sam",       "Seth",
+        4, "2004-10-10", "2009-08-27",      "Craig",       "Khai",
+        5, "2000-12-05", "2005-02-28",     "Parker",     "Gracie",
```
Comment (DavisVaughan) [inline data-raw/family.R:9]: I noticed that `family` is a double column, not sure if you wanted an integer column. Probably doesn't matter but thought id point it out

File: data-raw/family.R [hunk 2]
```diff
@@ -0,0 +1,15 @@
+library(dplyr)
```
Comment (DavisVaughan) [inline data-raw/family.R:1]:
```suggestion
library(tidyverse)
```

because of `parse_date` used below

File: data-raw/family.R [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        2, "1996-06-22",           NA,       "Mark",           NA,
+        3, "2002-07-11", "2004-04-05",        "Sam",       "Seth",
+        4, "2004-10-10", "2009-08-27",      "Craig",       "Khai",
+        5, "2000-12-05", "2005-02-28",     "Parker",     "Gracie",
+)
+family <- family |>
+  mutate(across(starts_with("dob"), parse_date))
+family
+
+usethis::use_data(family, overwrite = TRUE)
```
Comment (DavisVaughan) [inline data-raw/family.R:15]: For some reason if I try and run this code then git tells me that `family.rda` has changed. Not sure if that is worrying or not.

Even if I load up tidyr and immediately resave the `family` dataset from tidyr, I get a diff. So I guess it has something to do with `use_data()`

Comment (DavisVaughan) [reply data-raw/family.R:15]: This is true, so it makes me feel a little better, still not sure what is happening when I save it though

``` r
library(tidyverse)
#> Warning: package 'readr' was built under R version 4.1.2
#> Warning: package 'dplyr' was built under R version 4.1.2

family2 <- tribble(
  ~family,  ~dob_child1,  ~dob_child2, ~name_child1, ~name_child2,
  1, "1998-11-26", "2000-01-29",      "Susan",       "Jose",
  2, "1996-06-22",           NA,       "Mark",           NA,
  3, "2002-07-11", "2004-04-05",        "Sam",       "Seth",
  4, "2004-10-10", "2009-08-27",      "Craig",       "Khai",
  5, "2000-12-05", "2005-02-28",     "Parker",     "Gracie",
)
family2 <- family2 |>
  mutate(across(starts_with("dob"), parse_date))

identical(family, family2)
#> [1] TRUE
```

<sup>Created on 2022-03-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

File: data-raw/who.csv [hunk 4]
```diff
@@ -4368,40 +4368,40 @@ Myanmar,MM,MMR,2010,106,3043,6578,6688,5607,3632,2308,196,2452,3454,2752,2525,18
...
-Namibia,"NA",NAM,2010,36,359,852,680,287,146,126,67,429,685,382,206,122,87,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA
-Namibia,"NA",NAM,2011,48,337,844,660,361,152,138,78,427,653,410,185,100,110,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA
-Namibia,"NA",NAM,2012,61,358,810,686,292,157,137,81,394,582,396,198,84,97,386,153,477,516,268,162,135,339,200,379,306,183,98,90,180,126,228,261,138,46,55,163,116,219,177,84,25,47,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA
-Namibia,"NA",NAM,2013,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,603,612,1496,1404,723,392,309,491,688,1137,747,408,220,202
+Namibia,NA,"NAM",1980,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA,NA
```
Comment (DavisVaughan) [inline data-raw/who.csv:4371]: This is normal?

**Review by DavisVaughan — APPROVED** [review]

File: data-raw/who.csv (hunk 4 above)
Comment (hadley) [reply data-raw/who.csv:4371]: I suspect this is just a change in how readr writes NA; I think were some changes around whether `"NA"` was interpreted as missing or not.

File: data-raw/family.R (hunk 3 above)
Comment (hadley) [reply data-raw/family.R:15]: Yeah, I noticed that too. I have a vague recollection that maybe rda includes a time stamp?

File: data-raw/who.csv (hunk 4 above)
Comment (DavisVaughan) [reply data-raw/who.csv:4371]: It also struck me as weird that `"NAM"` started getting quoted all of a sudden

File: data-raw/family.R (hunk 3 above)
Comment (DavisVaughan) [reply data-raw/family.R:15]: At the very least, it adds the R version in there, which likely explains our differences
https://github.com/wch/r-source/blob/6b22b60126646714e0f25143ac679240be251dbe/src/main/serialize.c#L1397

File: R/data.R [hunk 5]
```diff
@@ -173,3 +184,83 @@
...
+#'
+#' @format A data frame with 5 rows and 5 columns:
+#' \describe{
+#'   \item{family}{Family identifier}
+#'   \item{dob_child1}{Date of birth of first child}
+#'   \item{dob_child2}{Date of birth of second child}
+#'   \item{name_child1}{Name of first child}?
+#'   \item{name_child2}{Name of second child}
+#' }
+"family"
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/data.R:202]: Hmmm, maybe `untidy_family`?

File: data-raw/who.csv (hunk 4 above)
Comment (hadley) [reply data-raw/who.csv:4371]: Yeah, looks like maybe a small readr bug: https://github.com/tidyverse/readr/issues/1386

File: R/data.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/data.R:202]: Or `household`

[the remark this answers is not in the document]
Comment (hadley) [reply R/data.R:202]: Oh I like `household`

**Pull request merged**